### PR TITLE
Adapt protocol.txt to md format

### PR DESCRIPTION
**Pull request by Chemaclass**
Adapt protocol.txt to md format

### CODE REVIEW

**Comment by sergeyklay** [comment]
https://github.com/kr/beanstalkd/pull/308

**Comment by Chemaclass** [comment]
@sergeyklay I didn't know this. Thanks for your answer :)

File: doc/protocol.md [hunk 1]
```diff
@@ -34,20 +34,20 @@ If a client violates the protocol (such as by sending a request that is not
...
 
- - "BAD_FORMAT\r\n" The client sent a command line that was not well-formed.
-   This can happen if the line does not end with \r\n, if non-numeric
+ - "BAD_FORMAT" The client sent a command line that was not well-formed.
+   This can happen if the line does not end with , if non-numeric
    characters occur where an integer is expected, if the wrong number of
    arguments are present, or if the command line is mal-formed in any other
    way.
 
- - "UNKNOWN_COMMAND\r\n" The client sent a command that the server does not
```
Comment (JensRantil) [inline doc/protocol.md]: `\r\n` is relevant for protocol implementors. Must be kept.

File: doc/protocol.md [hunk 2]
```diff
@@ -71,21 +71,11 @@ will execute the job; when it is finished the worker will send a "delete"
 command to delete the job.
 
 Here is a picture of the typical job lifecycle:
-
-
-   put            reserve               delete
-  -----> [READY] ---------> [RESERVED] --------> *poof*
```
Comment (JensRantil) [inline doc/protocol.md]: Why was this removed?

**Comment by JensRantil** [comment]
I'll be closing this. Let me know if you disagree.
